pkg/chunk/series_store.go

```diff
@@ -281,10 +282,25 @@ func (c *seriesStore) lookupSeriesByMetricNameMatchers(ctx context.Context, from
 		return nil, err
 	}
 
+	level.Debug(log).Log("msg", "lookupSeriesByMetricNameMatchers:shard", "shard", shard, "label_index", shardLabelIndex, "matchers", len(matchers))
+
 	if shard != nil {
 		matchers = append(matchers[:shardLabelIndex], matchers[shardLabelIndex+1:]...)
+		// Just get series for metric if there are no matchers
```

Comment:
we'd still need to filter the resulting seriesIDs here. I think it can more be more succinctly expressed by calculating the shard/splicing shard labels then running the `len(matchers) == 0` logic once.